pyodk/endpoints/submissions.py

```diff
@@ -159,12 +153,165 @@ def get_table(
+        project_id: Optional[int] = None,
+        device_id: Optional[str] = None,
+    ) -> Submission:
+        """
+        Create a Submission.
+
+        Example submission XML structure:
+
+        <data id="my_form" version="v1">
+          <meta>
+            <instanceID>uuid:85cb9aff-005e-4edd-9739-dc9c1a829c44</instanceID>
+          </meta>
+          <name>Alice</name>
+          <age>36</age>
+        </data>
+
+        :param xml: The submission XML.
+        :param form_id: The xmlFormId of the Form being referenced.
+        :param project_id: The id of the project this form belongs to.
+        :param device_id: An optional deviceID associated with the submission.
+        """
+        try:
+            pid = pv.validate_project_id(project_id, self.default_project_id)
+            fid = pv.validate_form_id(form_id, self.default_form_id)
+            params = {}
+            if device_id is not None:
+                params["deviceID"] = device_id
+        except PyODKError as err:
+            log.error(err, exc_info=True)
+            raise err
+
+        response = self.session.response_or_error(
+            method="POST",
+            url=self.urls.post.format(project_id=pid, form_id=fid),
+            logger=log,
+            headers={"Content-Type": "application/xml"},
+            params=params,
+            data=xml,
+        )
+        data = response.json()
+        return Submission(**data)
+
+    def put(
+        self,
+        instance_id: str,
+        xml: str,
+        form_id: Optional[str] = None,
+        project_id: Optional[int] = None,
+    ) -> Submission:
+        """
+        Update Submission data.
+
+        Example submission XML structure:
+
+        <data id="my_form" version="v1">
+          <meta>
+            <deprecatedID>uuid:85cb9aff-005e-4edd-9739-dc9c1a829c44</deprecatedID>
+            <instanceID>uuid:315c2f74-c8fc-4606-ae3f-22f8983e441e</instanceID>
+          </meta>
+          <name>Alice</name>
+          <age>36</age>
+        </data>
+
+        :param instance_id: The instanceId of the Submission being referenced.
+        :param xml: The submission XML.
+        :param form_id: The xmlFormId of the Form being referenced.
+        :param project_id: The id of the project this form belongs to.
+        """
+        try:
+            pid = pv.validate_project_id(project_id, self.default_project_id)
+            fid = pv.validate_form_id(form_id, self.default_form_id)
+            iid = pv.validate_instance_id(instance_id)
+        except PyODKError as err:
+            log.error(err, exc_info=True)
+            raise err
+
+        response = self.session.response_or_error(
+            method="PUT",
+            url=self.urls.put.format(project_id=pid, form_id=fid, instance_id=iid),
+            logger=log,
+            headers={"Content-Type": "application/xml"},
+            data=xml,
+        )
+        data = response.json()
+        return Submission(**data)
+
+    def patch(
```

Comment:
`review`? `update_review_state`? I think even if the `patch` verb does eventually allow for updating more metadata we'll want to keep the ability to just update the review state.

For convenience, this method should also take a comment string parameter. Another option would be to make this method "private" and expose a separate `review`. That would match what I've proposed for `edit`/`put`.

There are some nice workflows that can be automated with this. For example, I've seen scripts that identify outlier submissions and change their review state to "has issues" with a useful comment. Then someone can use the frontend to review issues and act on them.

Reply:
I'm OK with the name as-is but please let me know if you'd like to change it. Please confirm if you'd like a `comment` parameter in Submission.patch? Presumably this means review state changes accept a comment in the UI.

Reply:
_patch marked private and `review` method added with comment param